homeassistant/components/rmvtransport/config_flow.py

```diff
@@ -0,0 +1,146 @@
+"""Config flow to configure the Rhein-Main public transport component."""
+from collections import OrderedDict
+import hashlib
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_SCAN_INTERVAL, CONF_SHOW_ON_MAP
+from homeassistant.helpers import aiohttp_client
+
+from .const import (
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    CONF_HASH,
+    CONF_STATION,
+    CONF_DESTINATIONS,
+    CONF_DIRECTION,
+    CONF_LINES,
+    CONF_PRODUCTS,
+    CONF_TIME_OFFSET,
+    CONF_MAX_JOURNEYS,
+    VALID_PRODUCTS,
+)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class RMVTransportFlowHandler(config_entries.ConfigFlow):
+    """Handle a rmv transport config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the RMV transport config flow."""
+        self.rmv_config = {}
+
+    async def _show_form_lines(self, errors=None):
+        """Show the setup form to the user."""
+        data_schema = OrderedDict()
+        data_schema[vol.Optional(CONF_DESTINATIONS)] = str
+        data_schema[vol.Optional(CONF_LINES, default="")] = str
+        data_schema[vol.Optional(CONF_DIRECTION)] = str
+
+        return self.async_show_form(
+            step_id="user", data_schema=vol.Schema(data_schema), errors=errors or {}
+        )
+
+    async def _show_form_products(self, errors=None):
+        """Show the setup form to the user."""
+        data_schema = OrderedDict()
+
+        for product in VALID_PRODUCTS:
+            data_schema[vol.Optional(product, default=True)] = bool
+
+        return self.async_show_form(
+            step_id="user", data_schema=vol.Schema(data_schema), errors=errors or {}
+        )
+
+    async def _show_form_details(self, errors=None):
```

Comment:
These extra options should not be available in the config flow. We only want the critical options in the config flow. Extra options can be configured via configuration.yaml.